bpf/lib/fib.h

```diff
@@ -279,27 +211,31 @@ fib_lookup_v6(struct __ctx_buff *ctx, struct bpf_fib_lookup_padded *fib_params,
 
 static __always_inline int
 fib_redirect_v6(struct __ctx_buff *ctx, int l3_off,
-		struct ipv6hdr *ip6, const bool needs_l2_check,
-		__s8 *fib_err, int iif, int *oif)
+		struct ipv6hdr *ip6 __maybe_unused, const bool needs_l2_check,
+		__s8 *fib_err __maybe_unused, int *oif)
 {
-	struct bpf_fib_lookup_padded fib_params = {
-		.l = {
-			.family		= AF_INET6,
-			.ifindex	= iif,
-		},
-	};
+	struct bpf_fib_lookup_padded fib_params __maybe_unused = {0};
+	void *data_end;
+	void *data;
+	__s8 skip_fib __maybe_unused = BPF_FIB_LKUP_RET_NO_NEIGH;
 	int ret;
 
-	ipv6_addr_copy((union v6addr *)&fib_params.l.ipv6_src,
-		       (union v6addr *)&ip6->saddr);
-	ipv6_addr_copy((union v6addr *)&fib_params.l.ipv6_dst,
-		       (union v6addr *)&ip6->daddr);
-
 	ret = ipv6_l3(ctx, l3_off, NULL, NULL, METRIC_EGRESS);
 	if (unlikely(ret != CTX_ACT_OK))
 		return ret;
 
-	return fib_redirect(ctx, needs_l2_check, &fib_params, fib_err, oif);
+	/* we touched ttl above, revalidate for any header reads/writes */
+	if (!revalidate_data(ctx, &data, &data_end, &ip6))
+		return DROP_INVALID;
+
```

Comment:
This shouldn't be required (there's no csum adjustment in the IPv6 case) ?

Reply:
Yup I believe you're right, we aren't touching packet structure. I'll remove.

Reply:
I recently realized that `ipv6_dec_hoplimit()` is the *one* place where we use `BPF_F_RECOMPUTE_CSUM` though. So let's see :grin:.

Reply:
Ah good point, but I guess if we reshuffle the code anyway, this can go away. We can reshuffle in the same manor as the ipv4 code you outlined above.

Reply:
Implemented, lets see how the tests go.